time.js

```diff
@@ -8,21 +8,321 @@
+    else if(when instanceof ohItem) {
+        return convertItem(items.getItem(when.name));
+    }
+
+    // items.Item
+    else if(when.constructor.name === 'Item') {
+        return convertItem(when);
+    }
+
+    // Unsupported
+    throw Error('"' + when + '" is an unsupported type for conversion to time.ZonedDateTime');
+
+}
+
+//openHAB uses a RFC DateTime string, js-joda defaults to the ISO version, this defaults RFC instead
 const rfcFormatter = time.DateTimeFormatter.ofPattern("uuuu-MM-dd'T'HH:mm:ss.SSS[xxxx][xxxxx]");
 const targetParse = time.ZonedDateTime.prototype.parse;
 time.ZonedDateTime.prototype.parse = function (text, formatter = rfcFormatter) {
-  return targetParse(text, formatter);
-};
+    return targetParse(text, formatter);
+}
 
 /**
- * Parses a ZonedDateTime to milliseconds from now until the ZonedDateTime.
- *
- * @memberof time
- * @returns {number} duration from now to the ZonedDateTime in milliseconds
+ * Moves the date portion of the date time to today, accounting for DST
+ */
+ time.ZonedDateTime.prototype.toToday = function () {
+    let now = time.ZonedDateTime.now();
+    return this.withYear(now.year())
+               .withMonth(now.month())
+               .withDayOfMonth(now.dayOfMonth());
+}
+
+/**
+ * Compares this ZDT to see if it falls between start and end times,
+ * accounting for times that span midnight. start and end can be any type
+ * that has a `toZonedDateTime()` method.
+ * time.toZDT is called on the arguments. Examples:
+ * 
+ * time.ZonedDateTime.now().betweenTimes(items.getItem('Sunset'), "23:30:00") // is now between sunset and 11:00 pm?
+ * time.toZDT(items.geItem('MyDateTimeItem')).betweenTimes(time.toZDT(), time.toZDT(1000)) // is the state of MyDateTimeItem between now and a second from now?
+ * 
+ * @param {*} start the starting time in anything that can be 
+ * @param {*} end the ending time
+ * @returns {Boolean} true if this is between start and end
  */
-time.ZonedDateTime.prototype.millisFromNow = function () {
-  return time.Duration.between(time.ZonedDateTime.now(), this).toMillis();
-};
+ time.ZonedDateTime.prototype.betweenTimes = function(start, end) {
+    startTime = time.toZDT(start).toLocalTime();
+    endTime = time.toZDT(end).toLocalTime();
+    currTime = this.toLocalTime();
 
+    // time range spans midnight
+    if(endTime.isBefore(startTime)) {
+        return currTime.isAfter(startTime) || currTime.isBefore(endTime);
+    }
+    else {
+        return currTime.isAfter(startTime) && currTime.isBefore(endTime);
+    }
+} 
+ 
+/**
+ * Tests to see if the difference betwee this and the passed in ZoneDateTime is
+ * within the passed in maxDur.
+ * @param {time.ZonedDateTime} zdt the date tiem to compare to this
+ * @param {time.Duration} maxDur the duration to test that the difference between this and zdt is within
+ * @returns {Boolean} true if the delta between this and zdt is within maxDur
+ */
+time.ZonedDateTime.prototype.isClose = function(zdt, maxDur) {
```

Comment:
I'm struggling with this method a bit, it seems pretty hard to explain (maybe is needs a better name and better named parameters?) , and also seems very niche in its use case. I'm wondering how frequently something like this will be used by others?  I think we should strive to only add in whats really going to be useful to a broader audience, otherwise our library will start to seem inconsistent and unfocused.

Reply:
I've seen a number of users struggle with the fact that you'll probably never end up with two timestamps that are actually equal. They are surprised that something like `time.ZonedDateTime.now() == time.ZonedDateTime.now()` will be false. This was. an attempt to give them a function I could point them to to help them make such similar comparisons as well as kind of explain what's going on. 

It's also handy in cases where you want to see how far apart two time stamps are (for example, if a motion detector triggered within five minutes of a light switch flipping).

I use the heck out of this function in the tests and originally it was only in my test code. But I ran into a few use cases where it seemed  useful in general. I can definitely move it to the tests, though will probably also reimplement it in my openhab-rules-tools since I've already two rules I've identified that can benefit from it.

Reply:
I haven't removed this function. I personally find it useful. But if you feel strongly about it I will remove it.